src/app/sample/sample-list/sample-list.component.ts

```diff
@@ -24,6 +24,7 @@ export class SampleListComponent implements OnInit {
         this.totalSamples = samples.length;
 
         // Get the related field data.
+        // Doesn't work
```

Comment:
Why do you say this doesn't work? If you associate a metadata value to the sample, its ID is shown properly in the Samples List

Reply:
the function does not get the metadataValue what it gets is an empty array

Reply:
Have you tried to associate metadataValues to the sample? If you do so, the array is not empty, it contains the associated values

Reply:
Now I am testing the association through the API with the use of postman associating a sample to a metadataValue I get the following error:

"cause": {
         "cause": null,
         "message": "Can not deserialize instance of java.util.List out of VALUE_STRING token \ n at [Source: org.apache.catalina.connector.CoyoteInputStream@6ab0ec71; line: 3, column: 11] (through reference chain: cat.udl.eps.entsoftarch.textannot.domain.Sample [\ "has \"]) "
     },
     "message": "JSON parse error: Can not deserialize instance of java.util.List out of VALUE_STRING token; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.util.List out of VALUE_STRING token \ n at [Source: org.apache.catalina.connector.CoyoteInputStream@6ab0ec71; line: 3, column: 11] (through reference chain: cat.udl.eps.entsoftarch.textannot.domain.Sample [\ "has \ "])"

Reply:
To create a metadataValue associated to a sample do like in the API tests. POST to /metadataValue to create the new one, and include "forA": "/samples/SAMPLE_ID"